docs/specs/code/MULTI_CURRENCY.md

```diff
@@ -0,0 +1,55 @@
+# Abstract
+
+Describe the ability to see and input ADA amount in another currency.
+
+# Proposal
+
+This feature includes two aspects:
+
+## 1. Input
+When the UI contains an ADA amount input, it should let the user input the number in another currency and covert it to ADA amount at the present conversion rate.
+
+## 2. Output
+When the UI shows an ADA amount, additionally it should be able to show the amount in another currency at a relevant conversion rate.
+
+# Risk
+
+This feature does open new attack vector, for example: an attacker can engage a transaction with a victim and by hacking the backend and manipulate the conversion rate, trick the victim into transferring more Ada than the intended amount.
```

Comment:
I would like to do the following:

- have an external service that is also getting the prices directly from the service providers
- pinging out API to get the prices. If there is a mismatch, we are in trouble and we can act quickly

Reply:
or also we could have services posting signed messages to our DB which could run independly 🤔

Reply:
The first comment sounds a good idea. Will implement.  I don't quite understand the second comment ("or also ..."). Could you please elaborate a bit?

Reply:
- that we could have independent services (different providers and architectures) that connect to the data pricing feeds
- every one of these independent services could have their own signing keys
- they could be non-read from the external world, so in order to share their data, they could write directly to our DB (their ip could be whitelisted from the DB’s side).
- once the prices are in our DB and signed, everything after that in the process should be sort of secure, not in the way that can’t be modified, but rather than any modification will be obvious and straight forward to notice because it won’t have the signed prices with the right key

Reply:
cc @vantuz-subhuman @rcmorano @rooooooooob feel free to call me out on this one if it doesn’t make sense for you guys